### PR TITLE
Feature/admin crud

### PR DESCRIPTION
**Pull request by oyakovytskyi**
## Describe your changes

- Component Refactoring: extracted the logic for create, update, and delete actions into dedicated components to enhance modularity.
- TypeScript Enhancements: added TypeScript interfaces for the adminList page for better type safety and code clarity.
- Localization: Integrated i18n for all text content added.
- API Organization: moved all API-related logic into a single dedicated file for cleaner separation of concerns.
- Toast Notifications: added a ToastContainer to the main application layout and use it for better UX
- General Refactoring: cleaned up and optimized the code.

## Issue ticket code (and/or) and link

- [Link to JIRA ticket][(#https://ticket-url](https://codecraftersintership.atlassian.net/jira/software/projects/SCRUM/boards/1?selectedIssue=SCRUM-62))

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

Delete
photo_2024-11-19_01-08-28
photo_2024-11-19_01-08-45

Add
photo_2024-11-19_01-08-48
photo_2024-11-19_01-08-51

Update
photo_2024-11-19_01-08-54
photo_2024-11-19_01-08-56
photo_2024-11-19_01-10-27

### CODE REVIEW

File: src/api/adminActions.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+};
+
+export const addAdmin = async (
+  formData: AdminForm,
+  companyId: number
+): Promise<Admin> => {
+  try {
+    const response = await axiosInstance.post(`/admins`, {
+      ...formData,
+      logo: 'http://logo-sample',
```
Comment (lyudad) [inline src/api/adminActions.ts]: 1.1.11

File: src/api/adminActions.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+};
+
+export const updateAdmin = async (
+  formData: AdminForm,
+  userId: number
+): Promise<Admin> => {
+  try {
+    const response = await axiosInstance.patch(`/admins/${userId}`, {
+      ...formData,
+      logo: 'http://logo-sample',
```
Comment (lyudad) [inline src/api/adminActions.ts]: 1.1.11

File: src/pages/components/AddNewAdmin/index.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+          helperText={errors.full_name?.message}
+        />
+        <TextInput
+          label={t('addNewAdmin.email')}
+          sx={input}
+          inputProps={{
+            ...register('email', {
+              required: t('addNewAdmin.errors.emptyEmail'),
+              pattern: {
+                value: /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/,
```
Comment (lyudad) [inline src/pages/components/AddNewAdmin/index.tsx]: you use it 2ce, its good sind that you need move it to separate constant and reuse in both files. no duplication

File: src/pages/components/AddNewAdmin/useAddNewAdmin.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+  const sendData = async (formData: AdminForm): Promise<void> => {
+    setIsSubmitting(true);
+    try {
+      await addAdmin(formData, companyId);
+      await refreshAdmins();
+      closeModal();
+      reset();
+    } catch (error) {
+      console.error('Error adding admin:', error);
```
Comment (lyudad) [inline src/pages/components/AddNewAdmin/useAddNewAdmin.ts]: no console, if its error - show toast

File: src/pages/components/DeleteAdmin/useDeleteAdmin.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  const handleOpen = (): void => setIsOpened(true);
+  const handleClose = (): void => setIsOpened(false);
+
+  const deleteAdminRequest = async (): Promise<void> => {
+    try {
+      await deleteAdmin(adminId);
+      await refreshAdmins();
+      handleClose();
+    } catch (error) {
+      console.error('Failed to delete admin:', error);
```
Comment (lyudad) [inline src/pages/components/DeleteAdmin/useDeleteAdmin.ts]: same, no console. it should be error that user will see

File: src/pages/components/AddNewAdmin/index.tsx
Comment (lyudad) [inline src/pages/components/AddNewAdmin/index.tsx:1]: add admin and edit admin are the same popups, why you create 2 separate coponents

Comment (oyakovytskyi) [reply src/pages/components/AddNewAdmin/index.tsx:1]: made it in one

File: src/pages/components/DeleteAdmin/useDeleteAdmin.ts (hunk 5 above)
Comment (oyakovytskyi) [reply src/pages/components/DeleteAdmin/useDeleteAdmin.ts]: okay

File: src/pages/components/AddNewAdmin/useAddNewAdmin.ts (hunk 4 above)
Comment (oyakovytskyi) [reply src/pages/components/AddNewAdmin/useAddNewAdmin.ts]: okay

File: src/pages/components/AddNewAdmin/index.tsx (hunk 3 above)
Comment (oyakovytskyi) [reply src/pages/components/AddNewAdmin/index.tsx]: understand

File: src/api/adminActions.ts (hunk 1 above)
Comment (oyakovytskyi) [reply src/api/adminActions.ts]: will do

File: src/api/adminActions.ts (hunk 2 above)
Comment (oyakovytskyi) [reply src/api/adminActions.ts]: will do

**Pull request merged**